modules/shadertools/src/modules/lighting/gouraud-material/gouraud-material.ts

```diff
@@ -20,8 +20,8 @@ export type GouraudMaterialUniforms = {
 export const gouraudMaterial: ShaderModule<GouraudMaterialProps, GouraudMaterialUniforms> = {
   name: 'gouraudMaterial',
   // Note these are switched between phong and gouraud
-  vs: GOURAUD_VS,
-  fs: GOURAUD_FS,
+  vs: PHONG_FS.replace('phongMaterial', 'gouraudMaterial'),
+  fs: PHONG_VS.replace('phongMaterial', 'gouraudMaterial'),
```

Comment:
Hmm... Seems no way to do this that is completely palatable... 

Maybe just a single module `phongGouraudMaterial` with a switch?

Reply:
Given that Gouraud shading is still based on the Phong model, why don't we remove the Gouraud material altogether and provide `lighting_getLightColor` et al in `vs` **and** `fs`. Then the application code can choose whether to perform lighting at the vertex stage (Gouraud) or fragment (Phong). The UBO will always be called `phongMaterial` and we won't have to worry about which shader module we are using